### PR TITLE
Remove devise-encryptable dependency, change salting/peppering mechanism & expose Argon2 options

### PR DESCRIPTION
**Pull request by moritzhoeppner**
Hi @erdostom
Here's my pull request resolving #11. A couple of notes:
- The whole thing is obviously a breaking change, so we would need to release a new major version.
- Since the salting and peppering mechanism has changed, we need to migrate existing Argon2 hashes. My idea was to set `password_salt` to `nil` in order to signal which hashes are migrated. Don't know if that's the best solution. I described it in the README in more detail.
- The Travis-CI pipeline seems to be broken. Since I don't have any experience with travis (Do they even offer a free plan for open source projects? I couldn't find it.), I took the liberty to add a GitHub actions configuration. If you want to stay with travis, we can simply revert the commit.
- We had 'pry' in the Gemfile which doesn't seem to be an actual dev dependency, so I removed it.
- I'd suggest that before the release we test the gem against different versions of the dependencies. But that's a different issue so I left it out for the time being. But if you agree, I'm happy to prepare another pull request for this.

### CODE REVIEW

**Comment by erdostom** [comment]
Hi @moritzhoeppner 
Thank you so much for this! Give me a couple of days to review

File: spec/devise-argon2_spec.rb [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
   before do
     Devise.pepper = nil
+    Devise.argon2_options = { m_cost: 3, t_cost: 1, p_cost: 1 }
```
Comment (erdostom) [inline spec/devise-argon2_spec.rb]: Great idea to expose the options

**Comment by erdostom** [comment]
@moritzhoeppner thank you very much for this pull request!

It's great that you included a pathway not just for upgrading from v1, but also from bcrypt.

I agree with your decisions re setting `password_salt = nil` and also to migrate to Github Actions.

This is such a breaking change that I'm considering adding a postinstall message to the `gemspec`.

Also, since this is pretty much a rewrite of the entire gem, I invite you to open another PR and add yourself to the `Authors` section of the `gemspec`.

**Comment by erdostom** [comment]
@moritzhoeppner I added Rails / ActiveRecord + Mongoid versions to the test matrix and fixed the Mongoid tests (they were not actually running on Mongoid). Also added the postinstall message. Take a look at `master`, let me know if you think anything else is needed before releasing the v2.

**Comment by moritzhoeppner** [comment]
@erdostom Great, thanks for your feedback and the fixes! There are a few minor changes I'd suggest, maybe for the next patch version. So I'm going make two more pull requests :)

**Pull request merged**